lib/compile/index.ts

```diff
@@ -300,11 +300,12 @@ function getJsonPointer(
 ): SchemaEnv | undefined {
   if (parsedRef.fragment?.[0] !== "/") return
   for (const part of parsedRef.fragment.slice(1).split("/")) {
-    if (typeof schema == "boolean") return
-    schema = schema[unescapeFragment(part)]
-    if (schema === undefined) return
+    if (typeof schema === "boolean") return
+    const partSchema = schema[unescapeFragment(part)]
+    if (partSchema === undefined) return
+    schema = partSchema
     // TODO PREVENT_SCOPE_CHANGE could be defined in keyword def?
-    const schId = typeof schema == "object" && schema[this.opts.schemaId]
+    const schId = typeof schema === "object" && schema[this.opts.schemaId]
```

Comment:
eslint is fine with `==` in this particular context, it's like that everywhere.

Reply:
yeah this wasn't an eslint fix, just style fix. Is there a reason to use coercive equality here?

Reply:
nope, just nitpick. It's like that everywhere (and eslint is happy with it because typeof always returns the string, so it's never coercive... nvm :)

Reply:
I'm unsure how to interpret the nvm. Do you want me to change it back?

Reply:
nvm = "never mind" :)

Reply:
no need to revert